packages/@aws-cdk/aws-lambda/lib/event-source-mapping.ts

```diff
@@ -196,6 +212,9 @@ export class EventSourceMapping extends cdk.Resource implements IEventSourceMapp
       maximumRetryAttempts: props.retryAttempts,
       parallelizationFactor: props.parallelizationFactor,
       topics: props.kafkaTopic !== undefined ? [props.kafkaTopic] : undefined,
+      sourceAccessConfigurations: props.kafkaSecretArn !== undefined ? [{ type: 'SASL_SCRAM_512_AUTH', uri: props.kafkaSecretArn }] : undefined,
```

Comment:
I just noticed that this is a bit more complicated if you want to connect to a self hosted Kafka cluster, which is only accessible via a VPC.
`sourceAccessConfigurations` is a list of multiple things [then](https://docs.aws.amazon.com/lambda/latest/dg/API_SourceAccessConfiguration.html). I now wonder if `EventSourceMappingOptions` should just expose `sourceAccessConfigurations` directly rather than pretending to be Kafka specific. The same goes for `selfManagedEventSource`. It seems like these are meant to be extended in the future.
That would also solve the problem for the `kafkaSecret` because it would stay in the Kafka specific event source.
What are your thoughts @nija-at? I'll just try it out.

Reply:
I did it.